spring-kafka/src/test/java/org/springframework/kafka/support/serializer/RetryingDeserializerTests.java

```diff
@@ -54,6 +55,39 @@ void testRetry() {
 		rdes.close();
 	}
 
+	@Test
+	void retryDeserializerWithRecoveryCallback() throws InterruptedException {
+		NonRecoverableDeser delegate = new NonRecoverableDeser();
+		RetryingDeserializer<String> rdes = new RetryingDeserializer<>(delegate, new RetryTemplate());
```

Comment:
You also can just inject a lambda which would throw that exception.
I don't think it matters how many time we retry.
Only what is important for this test that `RecoveryCallback` is called.